lib/message-routing/chat-service-router.js

```diff
@@ -33,6 +33,7 @@ class ChatServiceRouter {
     this.bot.on('closed', () => {
       this.logger.info('Chat socket closed. Attempting to reconnect....');
       if (this.chatToConnectTo === 'dgg') {
+        this.messageRelay.relayMessageToListeners('pollstop', null);
```

Comment:
Isn't the `closed` event emitted when the chat bot has to reconnect for whatever reason, but the chat backend is still healthy? If so, you can wind up in a situation where the bot thinks a gulag is no longer active while a gulag poll is still up.